src/hooks/useSchemaValidator.ts

```diff
@@ -0,0 +1,218 @@
+import { useDatatype } from './useDatatype';
+import { useEffect, useMemo, useState } from 'react';
+import { useConfig } from '@openmrs/esm-framework';
+import { ConceptTrue, ConceptFalse } from '../constants';
+import { OHRIFormField, OHRIFormSchema } from '../api/types';
+
+export function useSchemaValidator(schema: OHRIFormSchema) {
+  //initializing state
+  const [errors, setErrors] = useState([]);
+  const [warnings, setWarnings] = useState([]);
+  const [fullArray, setFullArray] = useState<Array<OHRIFormField>>([]);
+  const [answersArray, setAnswersArray] = useState([]);
+  const [conceptSet, setConceptSet] = useState<Set<string>>(new Set());
+  const [answerConceptSet, setAnswerConceptSet] = useState<Set<string>>(new Set());
+  const { dataTypeToRenderingMap, conceptDataTypes } = useConfig();
```

Comment:
Have you actually tested this out? I don't see where you defined the config schema plus you would have to load this config by module name:

```jsx
const { dataTypeToRenderingMap, conceptDataTypes } = useConfig({
    externalModuleName: "@openmrs/openmrs-form-engine-lib",
  });

```